generators/kubernetes/templates/db/_kafka.yml

```diff
@@ -0,0 +1,62 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: <%= app.baseName.toLowerCase() %>-kafka
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: <%= app.baseName.toLowerCase() %>-kafka
+    spec:
+      containers:
+      - name: kafka
+        image: <%= DOCKER_KAFKA %>
+        env:
+        - name: KAFKA_ADVERTISED_HOST_NAME
+          value: localhost
```

Comment:
not sure about localhost, I think it should be:
`<%= app.baseName.toLowerCase() %>-kafka.<%= kubernetesNamespace %>.svc.cluster.local` instead of `localhost`

See my PR https://github.com/jhipster/generator-jhipster/pull/4364